invokeai/frontend/web/src/features/controlLayers/hooks/useExportCanvasToPSD.ts

```diff
@@ -0,0 +1,265 @@
+
+      // Validate canvas dimensions
+      if (canvasWidth <= 0 || canvasHeight <= 0) {
+        toast({
+          id: 'INVALID_CANVAS_DIMENSIONS',
+          title: t('toast.invalidCanvasDimensions'),
+          status: 'error',
+        });
+        return;
+      }
+
+      if (canvasWidth > 30000 || canvasHeight > 30000) {
+        toast({
+          id: 'CANVAS_TOO_LARGE',
+          title: t('toast.canvasTooLarge'),
+          description: t('toast.canvasTooLargeDesc'),
+          status: 'error',
+        });
+        return;
+      }
+
+      log.debug(`PSD canvas dimensions: ${canvasWidth}x${canvasHeight}`);
+
+      // Create PSD layers from active raster layers
+      const psdLayers = await Promise.all(
+        layerAdapters.map((layerData, index) => {
+          try {
+            const { adapter: _adapter, canvas, rect, layer } = layerData;
+
+            const layerDataPSD = {
+              name: layer.name || `Layer ${index + 1}`,
+              left: Math.floor(rect.x - minLeft),
+              top: Math.floor(rect.y - minTop),
+              right: Math.floor(rect.x - minLeft + rect.width),
+              bottom: Math.floor(rect.y - minTop + rect.height),
+              opacity: Math.floor(layer.opacity * 255),
+              hidden: false,
+              blendMode: 'normal' as const,
+              canvas: canvas,
+            };
+
+            log.debug(
+              `Layer "${layerDataPSD.name}": ${layerDataPSD.left},${layerDataPSD.top} to ${layerDataPSD.right},${layerDataPSD.bottom}`
+            );
+
+            return layerDataPSD;
+          } catch (error) {
+            log.error({ error: serializeError(error as Error) }, `Error processing layer ${layerData.layer.name}`);
+            return null;
+          }
+        })
+      );
+
+      // Filter out any failed layers
+      const validLayers = psdLayers.filter((layer) => layer !== null);
+
+      if (validLayers.length === 0) {
+        toast({
+          id: 'FAILED_TO_PROCESS_LAYERS',
+          title: t('toast.failedToProcessLayers'),
+          status: 'error',
+        });
+        return;
+      }
+
+      // Create PSD document
+      const psd = {
+        width: canvasWidth,
+        height: canvasHeight,
+        channels: 3, // RGB
+        bitsPerChannel: 8,
+        colorMode: 3, // RGB mode
+        children: validLayers,
+      };
+
+      log.debug(
+        {
+          layerCount: validLayers.length,
+          canvasDimensions: { width: canvasWidth, height: canvasHeight },
+          layers: validLayers.map((l) => ({
+            name: l.name,
+            bounds: { left: l.left, top: l.top, right: l.right, bottom: l.bottom },
+          })),
+        },
+        'Creating PSD with layers'
+      );
+
+      // Generate PSD file
+      const buffer = writePsd(psd);
+
+      // Create blob and download
+      const blob = new Blob([buffer], { type: 'application/octet-stream' });
+      const url = URL.createObjectURL(blob);
```

Comment:
Use `downloadBlob` from `invokeai/frontend/web/src/features/controlLayers/konva/util.ts` instead. You could update the utility to call `URL.revokeObjectURL(url);`, which is does not do right now but should do.